### PR TITLE
APIS-5668 Allow SDST users to grant request which has failed checks

### PR DESCRIPTION
**Pull request by petekirby-ee**

### CODE REVIEW

File: app/uk/gov/hmrc/apigatekeeperapprovalsfrontend/views/ProvideEscalatedToForGrantingPage.scala.html [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+  <link rel="stylesheet" href='@routes.Assets.versioned("stylesheets/application.css")'>
+
+  @ErrorSummary(form)
+  
+  @helper.form(action = uk.gov.hmrc.apigatekeeperapprovalsfrontend.controllers.routes.GrantedJourneyController.provideEscalatedToAction(model.applicationId), 'class -> "form") {
+    @helper.CSRF.formField
+
+    <span class="govuk-caption-l">@model.appName</span>
+    <h1 class="govuk-label-wrapper">
+      <label class="govuk-label govuk-label--l" for="first-name">
```
Comment (robdawson-hmrc) [inline app/uk/gov/hmrc/apigatekeeperapprovalsfrontend/views/ProvideEscalatedToForGrantingPage.scala.html:35]: is 'for="first-name"' needed here? I cant see a field with that name

File: app/uk/gov/hmrc/apigatekeeperapprovalsfrontend/views/ProvideEscalatedToForGrantingPage.scala.html [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+    </h1>
+    <div id="escalatedbyuser-hint" class="govuk-hint">
+      @{messages("escalatedtouser.note")}
+    </div>
+
+    <div class="govuk-form-group">
+      <label class="govuk-label" for="escalatedto">
+        @{messages("escalatedtouser.escalatedto")}
+      </label>
+      <input class="govuk-input govuk-input--width-10" id="escalatedto" name="escalatedto" type="text">
```
Comment (robdawson-hmrc) [inline app/uk/gov/hmrc/apigatekeeperapprovalsfrontend/views/ProvideEscalatedToForGrantingPage.scala.html:47]: should the form have 2 fields for the name first/last ?

File: app/uk/gov/hmrc/apigatekeeperapprovalsfrontend/controllers/GrantedJourneyController.scala [hunk 3]
```diff
@@ -91,6 +100,34 @@ class GrantedJourneyController @Inject()(
     GrantedJourneyController.provideWarningsForm.bindFromRequest.fold(handleInvalidForm, handleValidForm)
   }
 
+  def provideEscalatedToPage(applicationId: ApplicationId) = loggedInWithApplicationAndSubmission(applicationId) { implicit request =>
+    successful(Ok(provideEscalatedToForGrantingPage(provideEscalatedToForm, ViewModel(request.application.name, applicationId))))
+  }
+
+  def provideEscalatedToAction(applicationId: ApplicationId) = loggedInWithApplicationAndSubmission(applicationId) { implicit request =>
+    def handleValidForm(form: ProvideEscalatedToForm) = {
```
Comment (peteslater-ee) [inline app/uk/gov/hmrc/apigatekeeperapprovalsfrontend/controllers/GrantedJourneyController.scala:108]: I would question the use of EitherT here when we seem to only be handling a Future[Option]]

Wouldn't it be clearer to just map over the future and pattern match the option?

`
  def handleValidForm(form: ProvideEscalatedToForm) = {
      submissionReviewService
        .updateEscalatedTo(form.escalatedTo)(request.submission.id, request.submission.latestInstance.index)
        .map {
          case Some(value)  => Redirect(uk.gov.hmrc.apigatekeeperapprovalsfrontend.controllers.routes.GrantedJourneyController.provideWarningsPage(applicationId).url)
          case None         => {
            logger.warn(s"Error granting access for application $applicationId: There was a problem updating the grant warnings on the submission review")
            InternalServerError(errorHandler.internalServerErrorTemplate)
          }
        }
    }
`

File: app/uk/gov/hmrc/apigatekeeperapprovalsfrontend/views/ProvideEscalatedToForGrantingPage.scala.html (hunk 1 above)
Comment (petekirby-ee) [reply app/uk/gov/hmrc/apigatekeeperapprovalsfrontend/views/ProvideEscalatedToForGrantingPage.scala.html:35]: Fixed

File: app/uk/gov/hmrc/apigatekeeperapprovalsfrontend/views/ProvideEscalatedToForGrantingPage.scala.html (hunk 2 above)
Comment (petekirby-ee) [reply app/uk/gov/hmrc/apigatekeeperapprovalsfrontend/views/ProvideEscalatedToForGrantingPage.scala.html]: Yes, long story but fixed now....

File: app/uk/gov/hmrc/apigatekeeperapprovalsfrontend/controllers/GrantedJourneyController.scala (hunk 3 above)
Comment (petekirby-ee) [reply app/uk/gov/hmrc/apigatekeeperapprovalsfrontend/controllers/GrantedJourneyController.scala:109]: I agree - fixed

**Review by robdawson-hmrc — APPROVED** [review]

**Pull request merged**